endpoint/config.go

```diff
@@ -0,0 +1,199 @@
+package endpoint
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"sync"
+
+	"github.com/git-lfs/git-lfs/git"
+)
+
+const defaultRemote = "origin"
+
+type Config struct {
```

Comment:
I'm still not sold on having a per-package `*Config` type. I think that splitting out and removing stuff from the `*config.Configuration` type is really good, but I think that we should go one step further and figure out what an appropriate abstraction is.

What do you think about `*endpoint.Finder`?